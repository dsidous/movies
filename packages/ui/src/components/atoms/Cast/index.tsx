import * as React from 'react';
import { Link } from 'react-router-dom';

import { Box, Typography } from '@material-ui/core';

import { MediaImage } from '@atoms/index';
import { Cast_Crew_Credit } from '@typesRoots/movie';

import useStyles from './styles';

type Type = 'full' | 'top';

interface CastProps {
  cast: Cast_Crew_Credit;
  type: Type;
}

const Cast: React.FC<CastProps> = ({ cast, type }) => {
  const classes = useStyles();
  const mediaType = type === 'full' ? 'miniProfile' : 'profile';

  const getJobOrCharacter = () =>
    ('character' in cast && cast.character) || ('job' in cast && cast.job);

  return (
    <Box className={`${classes.root} ${type}`}>
      <Link to={`/person/${cast.id}`}>
        <MediaImage
          mediaType={mediaType}
          size={1}
          filePath={cast.profile_path}
          name={cast.name}
        />
        <Box className={classes.copy} py={1}>
          <Typography variant='subtitle2'>{cast.name}</Typography>
          <Typography variant='caption'>{getJobOrCharacter()}</Typography>
        </Box>
      </Link>
    </Box>
  );
};

export default Cast;
