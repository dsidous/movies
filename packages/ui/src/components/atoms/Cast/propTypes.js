import { object } from 'prop-types';

export const propTypes = {
  cast: object.isRequired,
};

export const defaultProps = {
  cast: {},
};
