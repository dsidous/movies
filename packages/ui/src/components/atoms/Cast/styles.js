import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  cast__element: {
    '&.top': {
      borderRadius: 4,
      boxShadow: '0 0 40px rgba(0,0,0,.1)',
      padding: '0 0 10px',
      textAlign: 'center',
      overflow: 'hidden',

      '& a': {
        color: '#666',
      },

      '& .cast__img': {
        cursor: 'pointer',
        margin: '0 0 10px',
        width: '100%',
      },
    },

    '&.full a': {
      color: '#666',
      cursor: 'pointer',
      display: 'grid',
      gridTemplateColumns: '1fr 2fr',
      gridGap: 20,
      padding: '10px 0',

      [theme.breakpoints.up('sm')]: {
        gridTemplateColumns: '1fr 4fr',
      },

      '& .cast__img': {
        borderRadius: 4,
        width: '100%',
      },
    },
  },

  cast__copy: {
    alignSelf: 'center',
  },
}));
