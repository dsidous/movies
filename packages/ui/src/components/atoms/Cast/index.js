import React from 'react';
import { Link } from 'react-router-dom';

import { propTypes, defaultProps } from './propTypes';
import MediaImage from '../MediaImage';

const Cast = ({ cast, type }) => {
  const mediaType = type === 'full' ? 'miniProfile' : 'profile';

  return (
    <div className={`cast__element ${type}`}>
      <Link to={`/person/${cast.id}`}>
        <MediaImage
          mediaType={mediaType}
          size={1}
          filePath={cast.profile_path}
          name={cast.name}
          className="cast__img"
        />
        <div className="cast__copy">
          <strong>{cast.name}</strong>
          <br />
          <small>{cast.character}</small>
        </div>
      </Link>
    </div>
  );
};

Cast.propTypes = propTypes;
Cast.defaultProps = defaultProps;

export default Cast;
