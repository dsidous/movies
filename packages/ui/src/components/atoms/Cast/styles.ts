import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      '& a': {
        display: 'block',
      },

      '& img': {
        width: '100%',
      },

      '&.top': {
        borderRadius: theme.shape.borderRadius,
        boxShadow: theme.shadows[12],
        textAlign: 'center',
        overflow: 'hidden',
      },

      '&.full a': {
        display: 'grid',
        gridTemplateColumns: '1fr 2fr',
        gridGap: theme.spacing(3),

        [theme.breakpoints.up('sm')]: {
          gridTemplateColumns: '1fr 4fr',
        },

        '& img': {
          borderRadius: theme.shape.borderRadius,
        },
      },
    },

    copy: {
      alignSelf: 'center',
    },
  }),
  { name: 'cast' },
);
