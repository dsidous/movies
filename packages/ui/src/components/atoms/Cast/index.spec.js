import React from 'react';
import { shallow } from 'enzyme';
import Cast from '.';

const defaultProps = {
  cast: {
    name: 'test',
    id: 1,
    character: 'test',
    profile_path: '/PSK6GmsVwdhqz9cd1lwzC6a7EA.jpg',
  },
  type: 'full',
};

describe('Atoms/Cast', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Cast {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
