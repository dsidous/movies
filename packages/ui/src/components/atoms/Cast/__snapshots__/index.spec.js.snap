// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Cast should render as expected 1`] = `ShallowWrapper {}`;
